File: backend/app/schemas/market.py
from pydantic import BaseModel


class PPIRecord(BaseModel):
    quarter: str
    index: float
    change_pct: float


class RentalRecord(BaseModel):
    quarter: str
    property_type: str
    region: str
    index: float


class VolumeRecord(BaseModel):
    month: str
    new_sale: int
    sub_sale: int
    resale: int


class VacancyRecord(BaseModel):
    quarter: str
    property_type: str
    vacancy_pct: float

File: frontend/src/api/market.ts
import { apiGet } from './client'

export interface PPIRecord {
  quarter: string
  index: number
  change_pct: number
}

export interface RentalRecord {
  quarter: string
  property_type: string
  region: string
  index: number
}

export interface VolumeRecord {
  month: string
  new_sale: number
  sub_sale: number
  resale: number
}

export interface VacancyRecord {
  quarter: string
  property_type: string
  vacancy_pct: number
}

export const getPPI = () => apiGet<PPIRecord[]>('/api/market/ppi')
export const getRental = () => apiGet<RentalRecord[]>('/api/market/rental')
export const getVolume = () => apiGet<VolumeRecord[]>('/api/market/volume')
export const getVacancy = () => apiGet<VacancyRecord[]>('/api/market/vacancy')
